plasmapy/utils/decorators/tests/test_checks.py

```diff
@@ -10,10 +11,1066 @@
 from plasmapy.utils.decorators.checks import (
     _check_quantity,
     _check_relativistic,
+    check_units,
+    check_values,
     check_quantity,
     check_relativistic,
+    CheckBase,
+    CheckUnits,
+    CheckValues,
 )
-from plasmapy.utils.exceptions import (RelativityWarning, RelativityError)
+from plasmapy.utils.exceptions import (PlasmaPyWarning,
+                                       RelativityWarning,
+                                       RelativityError)
+from types import LambdaType
+from typing import (Any, Dict)
+from unittest import mock
+
+
+# ----------------------------------------------------------------------------------------
+# Test Decorator class `CheckBase`
+# ----------------------------------------------------------------------------------------
+class TestCheckBase:
+    """
+    Test for decorator class :class:`~plasmapy.utils.decorators.checks.CheckBase`.
+    """
+    def test_for_members(self):
+        assert hasattr(CheckUnits, 'checks')
+
+    def test_checks(self):
+        _cases = [
+            {'input': (None, {'x': 1, 'y': 2}),
+             'output': {'x': 1, 'y': 2}},
+            {'input': (6, {'x': 1, 'y': 2}),
+             'output': {'x': 1, 'y': 2, 'checks_on_return': 6}},
+        ]
+        for case in _cases:
+            cb = CheckBase(checks_on_return=case['input'][0], **case['input'][1])
+            assert cb.checks == case['output']
+
+
+# ----------------------------------------------------------------------------------------
+# Test Decorator class `CheckValues` and decorator `check_values`
+# ----------------------------------------------------------------------------------------
+class TestCheckUnits:
+    """
+    Tests for decorator :func:`~plasmapy.utils.decorators.checks.check_units` and
+    decorator class :class:`~plasmapy.utils.decorators.checks.CheckUnits`.
+    """
+
+    check_defaults = CheckUnits._CheckUnits__check_defaults  # type: Dict[str, Any]
+
+    @staticmethod
+    def foo_no_anno(x, y):
+        return x + y
+
+    @staticmethod
+    def foo_partial_anno(x: u.Quantity, y: u.cm) -> u.Quantity:
+        return x.value + y.value
+
+    @staticmethod
+    def foo_return_anno(x, y) -> u.um:
+        return x.value + y.value
+
+    @staticmethod
+    def foo_stars(x: u.Quantity, *args, y=3 * u.cm, **kwargs):
+        return x.value + y.value
+
+    @staticmethod
+    def foo_with_none(x: u.Quantity, y: u.cm = None):
+        return x.value + y.value
+
+    def test_inheritance(self):
+        assert issubclass(CheckUnits, CheckBase)
```

Comment:
That is way more meticulous than I would have thought about if writing it on my own!